packages/api/endpoints/rules.js

```diff
@@ -60,26 +61,33 @@ async function get(req, res) {
  * @returns {Promise<Object>} the promise of express response object
  */
 async function post(req, res) {
-  try {
-    const data = req.body;
-    const name = data.name;
+  const {
+    model = new models.Rule(),
+    dbClient = await getKnexClient(),
+  } = req.testContext || {};
+
+  const rule = req.body || {};
+  const name = rule.name;
 
-    const model = new models.Rule();
+  if (await model.exists(name)) {
+    return res.boom.conflict(`A record already exists for ${name}`);
+  }
 
+  try {
+    const record = await model.create(rule);
     try {
-      await model.get({ name });
-      return res.boom.conflict(`A record already exists for ${name}`);
+      const postgresRecord = await translateApiRuleToPostgresRule(record, dbClient);
+      await dbClient.insert(postgresRecord, 'cumulus_id').into(tableNames.rules);
     } catch (error) {
-      if (error instanceof RecordDoesNotExist) {
-        const record = await model.create(data);
-
-        if (inTestMode()) {
-          await addToLocalES(record, indexRule);
-        }
-        return res.send({ message: 'Record saved', record });
-      }
+      await model.delete({ name });
```

Comment:
We should also use a transaction here instead of the try/catch with a model.delete

Reply:
[`ea9c778` (#2012)](https://github.com/nasa/cumulus/pull/2012/commits/ea9c7784b4d0e9841c8ebee053f05bbdcce89ce7)